scrunch/datasets.py

```diff
@@ -2365,6 +2369,37 @@ def replace_values(self, variables, filter=None, literal_subvar=False):
             return
         return resp
 
+    def replace_from_csv(self, filename, chunksize=1000, push_rows=True):
+        """
+        Given a csv file in the format:
+        id, var1_alias, var2_alias
+        1,  14,         15
+
+        where the first column is the Dataset PK
+
+        Replace the values of the matching id, for the given variables
+        in the Dataset using the /stream endpoint:
+
+        [{id: 1, var1_alias: 14, var2_alias: 15}, ...]
+        """
+        streaming_state = self.resource.body.get('streaming', 'no')
+        streaming_ds = self.make_streaming()
+        importer = pycrunch.importing.Importer()
+        df_chunks = pd.read_csv(
+            filename,
+            header=0,
+            chunksize=chunksize
+        )
+        for chunk in df_chunks:
+            # This is a trick to get rid of np.int64, which is not
+            # json serializable
+            stream = chunk.to_json(orient='records')
+            stream = json.loads(stream)
+            importer.stream_rows(self.resource, stream)
+            if push_rows:
+                streaming_ds.push_rows(chunksize)
+        self.edit(streaming=streaming_state)
```

Comment:
Here you would be switching the streaming state back to "`no`". I am not sure if it is possible to make this change. Can you confirm? I recall there being some limitations about which way the streaming states of datasets could go.

Reply:
I am checking, and you can only go back to "`no`" IFF there are 0 pending rows. So if you didn't push rows, you cannot make your ds go back to "no" and will be stuck as streaming until you push them. When do you need to not push them? Why is the flag there?

Reply:
Yes I realized we cannot move the streaming flag when there is pending batches, so I will force to push rows. Ultimately if this change fails, the user will be get an Exception with the message.